### PR TITLE
Automate embedded RHODS install

### PR DESCRIPTION
**Pull request by lugi0**
To run this test case in Jenkins, you should point to **an empty cluster** (no rhods installed)

The parameters you will need to change are:

**Disable** DEPLOY_RHODS_OPERATOR
Update the ODS_CI_RUN_SCRIPT_ARGS to:
```
--test-case ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot
```
If you want to install the latest build in the embedded channel in production, or
```
--test-case ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot --test-variable image_url:brew.registry.redhat.io/rh-osbs/iib:IIB_NUMBER
```
If you want to install a custom IIB, in which case the cluster will also need to have an ICSP  and pull secret set.

### CODE REVIEW

**Comment by lugi0** [comment]
Prod install verified with rhods-smoke #3308

File: ods_ci/tests/Resources/Files/operatorV2/embedded_cs.txt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+CatalogSource           rhods-catalog-dev
+ConfigMap               c831fa3421fa182ed3dd44418173698c8f1bc0581981d4ced92d2708b38cb67
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Files/operatorV2/embedded_cs.txt:2]: is this name going to remain fixed? it sounds like random

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    Install Embedded RHODS    ${cluster_type}     ${image_url}
+
+Verify Embedded RHODS Installation
+    [Documentation]
+    [Tags]
+    RHODS Embedded Verification
+
+
+*** Keywords ***
+Install Embedded RHODS
```
Comment (bdattoma) [inline ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot:33]: why not implementing in tasks like done for uninstall keyword?

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        Fail    Use a clean cluster to run this installation
+    END
+    IF    "${cluster_type}" == "selfmanaged"
+        ${file_path} =    Set Variable    ods_ci/tests/Resources/Files/operatorV2/
+        ${rc}  ${out} =    Run And Return Rc And Output    oc create ns ${OPERATOR_NAMESPACE}
+        IF    ${rc}!=0    Fail
+        Log    ${out}    console=yes
+        Copy File    source=${file_path}operatorgroup_template.yaml    destination=${file_path}operatorgroup_apply.yaml
+        Run    sed -i 's/<OPERATOR_NAMESPACE>/${OPERATOR_NAMESPACE}/' ${file_path}operatorgroup_apply.yaml
+        ${rc}  ${out} =    Run And Return Rc And Output   oc apply -f ${file_path}operatorgroup_apply.yaml
```
Comment (bdattoma) [inline ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot:46]: there is already a kw to create operator group `Create Operator Group`

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        IF    ${rc}!=0    Fail
+        Log    ${out}    console=yes
+        Remove File    ${file_path}operatorgroup_apply.yaml
+        ${image_url_bool} =    Evaluate    '${image_url}' == ''
+        IF  ${image_url_bool}
+            # Prod 2.1 build
+            Log    Installing prod 2.1 build    console=yes
+            Copy File    source=${file_path}subscription_template_21.yaml    destination=${file_path}subscription_apply.yaml  # robocop: disable
+            Run    sed -i 's/<OPERATOR_NAMESPACE>/${OPERATOR_NAMESPACE}/' ${file_path}subscription_apply.yaml
```
Comment (bdattoma) [inline ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot:55]: there is a kw that creates the subscription and the operator group (`Install ISV Operator From OperatorHub Via CLI`) That could be extended to add the config part needed for the single namespace parameter.

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    Process Resource List    filename_in=${filepath}embedded_runtime.txt
+    ...    filename_out=${filepath}embedded_processed_runtime.txt
+    ${expected} =    Get File    ${filepath}embedded_processed_expected.txt
+    ${actual} =    Get File    ${filepath}embedded_processed_runtime.txt
+    Remove File    ${filepath}embedded_processed_runtime.txt
+    Remove File    ${filepath}embedded_runtime.txt
+    Remove File    ${filepath}embedded_processed_expected.txt
+    Should Be Equal As Strings    ${expected}    ${actual}
+
+Namespace Should Not Exist
```
Comment (bdattoma) [inline ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot:125]: could you move this to a common resource file?

File: ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot (hunk 2 above)
Comment (bdattoma) [reply ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/133__embedded_operator.robot:33]: Ah I see now there are 2 keywords to delete RHODS v2, one for embedded channel explicitly. Could we merge the 2 kw?

**Review by jgarciao — APPROVED** [review]

**Review by bdattoma — APPROVED** [review]

**Pull request merged**